src/utils.rs

```diff
@@ -35,6 +36,62 @@ pub fn write_byte_string(f: &mut Formatter, byte_string: &[u8]) -> fmt::Result {
     Ok(())
 }
 
+pub(crate) struct MergeIter<It1, It2>
+where
+    It1: Iterator,
+    It2: Iterator<Item = It1::Item>,
+{
+    it1: Peekable<It1>,
+    it2: Peekable<It2>,
+}
```

Comment:
It would be great if you add several tests to check that iterator work as expected

Reply:
Added some unit tests

Reply:
Nice! @dralley , feel free to merge when you ready.